sdk/core/Azure.Core/src/Pipeline/RedirectPolicy.cs

```diff
@@ -79,8 +89,11 @@ internal async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPi
 
                 response.Dispose();
 
-                // Clear the authorization header.
-                request.Headers.Remove(HttpHeader.Names.Authorization);
+                // Clear the authorization header if necessary
+                if (_authorizationHeaderRemovalCallback.Invoke(message))
```

Comment:
Is it a concern that this callback could be used to mutate the HttpMessage? Is there a more constrained subset of info that we could pass to the callback instead?

Reply:
I don't think so - that is the user's choice and they can always mutate the message in a custom policy. If we only provided the Response, then we may get feedback that there is something from the request that is needed.